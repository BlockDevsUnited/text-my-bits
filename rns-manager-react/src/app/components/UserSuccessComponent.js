import React from 'react';
import propTypes from 'prop-types';
import { multilanguage } from 'redux-multilanguage';

import closeBlue from '../../assets/img/close-blue.svg';
import blueCheck from '../../assets/img/check-blue.svg';

const UserSuccessComponent = ({
  strings, title, message, handleCloseClick, address,
  visible,
}) => {
  if (!visible) {
    return (<></>);
  }

  return (
    <div className="success">
      <button
        type="button"
        className="close"
        onClick={handleCloseClick}
      >
        <img src={closeBlue} alt={strings.close} />
      </button>
      <p>
        <img src={blueCheck} alt="" />
      </p>
      <p><strong>{title}</strong></p>
      <p>{message}</p>
      {address && (
        <p className="explorer">
          <a
            href={`${process.env.REACT_APP_BLOCK_EXPLORER}/tx/${address}`}
            target="_blank"
            rel="noopener noreferrer"
          >
            {strings.view_explorer}
          </a>
        </p>
      )}
    </div>
  );
};

UserSuccessComponent.defaultProps = {
  title: 'Success',
  message: 'Success Message',
  handleCloseClick: () => {},
  address: '',
  visible: true,
};

UserSuccessComponent.propTypes = {
  strings: propTypes.shape({
    close: propTypes.string.isRequired,
    view_explorer: propTypes.string.isRequired,
  }).isRequired,
  title: propTypes.string,
  message: propTypes.string,
  handleCloseClick: propTypes.func,
  address: propTypes.string,
  visible: propTypes.bool,
};

export default multilanguage(UserSuccessComponent);
