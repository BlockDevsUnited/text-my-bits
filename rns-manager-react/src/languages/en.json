{
  "home_title": "rif name service",
  "home_subtitle": "get your domain, forget your address",
  "register_your_domain": "register your domain",
  "what_is_rns_title": "what is rns?",
  "what_is_rns_text": "RIF Name Service provides an architecture which enables the identification of blockchain addresses by human-readable names.",
  "resolve_a_domain": "resolve a domain",
  "admin_your_domain_title": "admin your domain",
  "admin_your_domain_action_1": "admin domain",
  "admin_your_domain_action_2": "admin domain reolution",
  "admin_your_domain_action_3": "create subdomains",
  "admin": "admin",
  "develop": "develop",
  "libs": "libs",
  "docs": "docs",
  "github": "github",
  "learn": "learn",
  "rif_white_paper": "rif white paper",
  "rns_white_paper": "rns white paper",
  "search": "search",
  "resolve": "resolve",
  "invalid_name": "Invalid name. Must be lower case characters and/or numbers",
  "open": "open",
  "auction": "auction",
  "finalize": "finalize",
  "reveal": "reveal",
  "owned": "owned",
  "search_another_domain": "search another domain",
  "bid_in_auction": "bid in the domain auction",
  "finalize_the_auction": "finalize the auction",
  "reveal_your_bid": "reveal your bid",
  "learn_about_process": "learn about process",
  "hide": "hide",
  "process": "Any user can register any available domain name. Registering a name requires to complete 3 steps",
  "process_step_1": "Request to register",
  "process_step_1_explanation": "Your wallet will open and you will be asked to confirm the first of two transactions required for registration.",
  "process_step_2": "Wait for a minute",
  "process_step_3": "Register",
  "process_step_3_explanation": "Click ‘register’ and your wallet will re-open. Only after the 2nd transaction is confirmed you'll know if you got the domain",
  "learn_more": "learn more",
  "mycrypto_search_title": "Search status on MyCrypto",
  "mycrypto_select_network": "Select RSK network on the top right selector.",
  "mycrypto_go_to_interact": "Go to My Crypto contract interaction tab.",
  "mycrypto_select_registrar": "Select RNS Registrar contract on Existing Contract selector.",
  "mycrypto_access": "Access!",
  "mycrypto_on_read_write_select": "On Read/Write Contract select",
  "my_crypto_copy_paste_on": "Copy and paste this hash on",
  "mycrypto_read": "Read!",
  "result": "Result",
  "state": "State",
  "next_action": "Next action",
  "go_to_mycrypto": "Go to MyCrypto",
  "start_an_auction_for": "start an auction for",
  "start_auction": "start auction",
  "mycrypto_start_title": "Start auction in MyCrypto",
  "mycrypto_choose_checkout": "Choose your checkout method.",
  "mycrypto_check_gas": "Check gas according to",
  "mycrypto_write": "Write!",
  "bid_for": "bid for",
  "how_much_do_you_value": "how much do you value",
  "bid_error_one_rif": "You must bid at least 1 RIF",
  "secret_phrase": "secret phrase",
  "random": "random",
  "advanced_options": "advanced options",
  "bid": "bid",
  "mycrypto_bid_title": "Bid in MyCrypto",
  "mycrypto_seal_the_bid": "First we are going to seal the bid",
  "mycrypto_copy_paste_values": "Copy this values and paste them in MyCrypto fields",
  "mycrypto_bid_select_owner": "the domain's owner. You may want to use your wallet address.",
  "mycrypto_remember_to_save_bid_data": "Remember you must save all this data to unseal the bid on reveal phase!",
  "mycrypto_copy_response": "Copy the response here",
  "mycrypto_select_rif": "Select RIF contract on Existing Contract selector.",
  "finalize_auction_for": "finalize auction for",
  "finalize_auction": "finalize auction",
  "mycrypto_finalize_title": "Finalize auction on MyCrypto",
  "unseal_bid_for": "unseal your bid for",
  "upload_bid_data": "upload bid data",
  "value": "value",
  "unseal_bid": "unseal bid",
  "mycrypto_unseal_title": "Unseal bid on MyCrypto",
  "mycrypto_admin_title": "admin your domain in MyCrypto",
  "name": "name",
  "admin_ownership": "admin ownership",
  "field": "field",
  "choose_": "choose...",
  "owner": "owner",
  "resolver": "resolver",
  "ttl": "tll",
  "get": "get",
  "set": "set",
  "admin_subdomain_ownership": "admin subdomain ownership",
  "subdomain": "subdomain",
  "admin_resolution": "admin resolution",
  "check_domain_resolver": "check domain resolver",
  "resolver_result": "resolver result",
  "public_resolver": "Public Resolver",
  "addr": "addr",
  "content": "content",
  "rsk_wallet_needed": "you need an RSK wallet to resolve, register and admin domains",
  "get_metamask": "get Metamask",
  "use_with_mycrypto": "use with MyCrypto",
  "connect_metamask_to_rsk" : "connect Metamask to RSK network",
  "connect": "connect",
  "done": "done",
  "subdomains": "subdomains",
  "Get": "Get",
  "on_mycrypto": "on MyCrypto",
  "mycrypto_select_registry": "Select RNS Registry contract on Existing Contract selector.",
  "Set": "Set",
  "mycrypto_select_resolver": "Select RNS Resolver contract on Existing Contract selector.",
  "mycrypto_select_multi_resolver": "Select Multi-crypto Resolver contract on Existing Contract selector.",
  "log_in": "log in",
  "address": "address",
  "network": "network",
  "not_domains_owner_message": "You are not the domains's owner.",
  "get_the_domain": "Get the domain!",
  "switch_to_mycrypto": "switch to MyCrypto",
  "register": "register",
  "please_log_in": "Please log in.",
  "switch_to_metamask": "switch to Metamask",
  "start": "start",
  "edit": "edit",
  "cancel": "cancel",
  "notifications": "notifications",
  "unread_notifications": "unread notifications",
  "Confirmed": "confirmed",
  "Waiting_for_confirmation_": "waiting for confirmation...",
  "set_owner": "set owner",
  "language": "language",
  "active_wallet": "active wallet",
  "setup_title": "Set up",
  "setup_download_metamask": "Download Metamask",
  "setup_connect_to_rsk": "Connect Metamask to RSK network:",
  "setup_open_network_selector": "Open network selector",
  "setup_select_custom_rpc": "Select 'Custom RPC'",
  "setup_select_advanced": "In new RPC url select 'Advanced options' and fill",
  "setup_save": "Save",
  "setup_add_rif": "Add RIF Token view (optional)",
  "setup_open_left_menu": "Open left menu",
  "setup_go_to_add_token": "Go to 'Add token'",
  "setup_fill_token_address_with_": "In custom token tab fill Token address with: ",
  "setup_next": "Next",
  "setup_add_tokens": "Add tokens",
  "notifications_start_auction_title": "auction started",
  "notifications_start_auction_action": "make your bid",
  "notifications_bid_title": "new bid",
  "notifications_bid_dont_forget": "don't forget to unseal the bid!",
  "notifications_bid_event_title": "Unseal bid for",
  "notifications_bid_download_message": "you will need the bid data to unseal it.",
  "download": "download data",
  "notifications_unseal_title": "Unsealed bid",
  "notifications_unseal_dont_forget": "don't forget to finelize the auction!",
  "notifications_unseal_event_title": "Finalize auction for",
  "notifications_finalize_title": "auction finalized",
  "notifications_finalize_action": "set addr resolution",
  "notifications_new_owner": "new owner",
  "notifications_new_resolver": "new resolver",
  "notifications_new_ttl": "new ttl",
  "notifications_new_subdomain_owner": "new subdomain owner",
  "notifications_new_addr": "new addr",
  "notifications_new_content": "new content",
  "rsk_address": "rsk address",
  "chain_address": "chain address",
  "bitcoin": "bitcoin (BTC)",
  "ethereum": "ethereum (ETH)",
  "bitcoin_cash": "bitcoin cash (BCH)",
  "litecoin": "litecoin (LTC)",
  "log_out": "log out",
  "notifications_new_chain_addr": "new chain addr",
  "rsk": "rsk (RBTC)",
  "suggested_subdomains": "wallet, donate, me, ...",
  "rsk_resolver": "rsk address resolver",
  "multi_chain_resolver": "multi crypto address resolver",
  "migrate_multichain_resolver_title": "New feature available!",
  "migrate_multichain_resolver_message": "Migrate to the new Multi Chain Resolver. It provides your domain of addresses for all assets, for example, your Bitcoin address.",
  "set_address": "set address",
  "setup_fill_with": "Fill with:",
  "select_chain": "select chain",
  "get_value_on_mycrypto": "get value on MyCrypto",
  "get_owner": "get owner",
  "reverse_resolution": "reverse resolution",
  "set_reverse": "set reverse",
  "has_reverse": "already set",
  "notifications_new_reverse_resolution": "new reverse resolution registered",
  "fifs_migration": "fifs registrar migration",
  "already_migrated": "your domain has been succesfully migrated to the fifs registrar",
  "fifs_subdomain": "subdomains don't need to be migrated",
  "fifs_migrate": "migrate to fifs registrar",
  "due_migration_date": "you have to migrate your domain to the new FIFS registrar by clicking on the button below. the due migration date is DD/MM/YYYY 12:00 AM UTC",
  "action_needed": "action needed",
  "notifications_migrated_domain": "your domain has been migrated to the new fifs registrar",
  "domain_not_available": "not available",
  "start_registration_for": "start the registration for",
  "rental_period": "rental period in years",
  "commit": "request for registration",
  "notifications_registrar_committed": "your domain has been requested to be registered",
  "notifications_registrar_revealed":  "your domain has been registered",
  "login_explanation": "now you can login with your just registered domain",
  "rskTld": ".rsk",
  "blocked_domain": "domains with less than 5 characters are blocked. stay tuned, we are going to release them soon",
  "transfer_domain": "transfer domain",
  "transfer": "transfer",
  "already_transferred": "your domain ownership has been succesfully transferred",
  "notifications_transferred_domain_token": "your domain ownership has been succesfully transferred",
  "renew_your_domain": "renew your domain",
  "your_domain_expires": "Your domain expires in ",
  "days": " days.",
  "not_registered_domain": "The domain has not been registered or has expired, you should register it before renewing.",
  "start_renovation_for": "start renovation for",
  "renew": "renew",
  "renew_domain_successfully": "the domain has been renewed succesfully",
  "discount": "50% discount per year from the third year",
  "rif_per_year": "RIF per year",
  "price": "price",
  "how_long_want_name": "For how long do you want your name",
  "not_enough_balance": "you don't have enough balance to perform this action",
  "click_here_not_enough_balance": "click here and check where you can buy more tokens",
  "no_wallet": "No wallet detected",
  "network_mismatch": "Network Mismatch",
  "connect_to_network": "Connect your wallet to",
  "current_connected": "Your wallet is currently connected to",
  "connected_successful": "Connected Successfully",
  "unknown_network": "Unknown Network",
  "unlock_wallet": "Please unlock your wallet",
  "string_resolver": "String resolver",
  "notifications_new_str": "new str",
  "admin_str_resolution": "admin str resolution",
  "address_formatted_incorrectly" : "Your address is not formatted for this network.",
  "address_formatted_incorrectly_description" : "Please check that the address is correct before continuing. If the address is correct, please use the button below to convert your address to lowercase and proceed.",
  "convert_to_lowercase": "Convert address to lowercase and continue.",
  "registering": "Registering",
  "request_domain": "Request domain",
  "register_domain": "Register domain",
  "login": "Login",
  "did_you_know": "Did you know that",
  "tip1": "You can use your domain in RIF Marketplace to provide your services.",
  "read_more": "Read more about it",
  "domain_requested": "Your domain has been requested",
  "to_register": "To finish registering",
  "admin_domain": "Admin my domain",
  "register_another_domain": "Register another domain",
  "wait_for_two_minutes": "Wait for aproximately two minutes",
  "wait_period_reason": "the waiting period is required to securely request your domain.",
  "click_register_domain": "click 'Register domain' and your wallet will re-open.",
  "your_domain_will_be_registered": "Your domain will be registered after the transaction is confirmed.",
  "wait_transation_confirmed": "Wait until the transaction is confirmed. It will take less than a minute.",
  "how_long_want_domain": "How long do you want your domain",
  "period_in_years": "period in years",
  "auto_address_setup": "Automatic address setup",
  "auto_address_explanation": "If you choose this option, your wallet address will automatically be set as the resolved address of your domain.",
  "notification_domain_ready_register": "Your domain is ready to be registered",
  "advanced": "Advanced",
  "basic": "Basic",
  "domain_info": "Domain info",
  "your_addresses": "Your addresses",
  "reverse": "Reverse",
  "copied": "copied",
  "copy_text": "Copy text",
  "transfer_warning": "Transfer the ownership of your domain to another address. Please be aware that once you transfer the ownership of your domain you will not be able to access and manage the domain anymore.",
  "expires_on": "Expires on",
  "renew_explanation": "Your wallet will open and you will be asked to confirm the first of two transactions required for renovation.",
  "renew_domain": "Renew Domain",
  "close": "Close",
  "renew_success": "Your domain has been renewed.",
  "success": "Success",
  "view_explorer": "View it on the explorer",
  "was_transfered": "was transfered",
  "register_new_domain": "Register a new domain",
  "login_another_domain": "Login to another domain",
  "upgrade_domain": "Upgrade domain",
  "upgrade": "upgrade",
  "upgrade_domain_explanation": "The domain upgrade offers new functionalities and dynamic renovation. The upgrade requires a single transaction with your wallet, then your domain will be upgraded automatically.",
  "faq": "Frequently Asked Questions",
  "type_sudomain": "Type your subdomain name",
  "type_owners_address": "Type owner's address",
  "create": "Create",
  "my_subdomains": "My subdomains",
  "subdomain_has_been_registered": "The subdomain has been registered",
  "submit": "Submit",
  "subdomain_owner_set": "Subdomain owner has been set.",
  "delete": "delete",
  "remove_subdomain_comfirm": "Are you sure you want to remove this subdomain?",
  "purpose_subdomains": "You can create free subdomains for different purposes (personal wallet, gaming, betting...)",
  "Name": "Name",
  "Owner": "Owner",
  "use_the_new_admin_to_transfer": "Use the new admin to transfer domains.",
  "add_new_address": "Add new address",
  "paste_your_address": "Paste your address",
  "add": "Add",
  "delete_chain_confirm": "Are you sure you want to remove this chain address?",
  "string_resolver_message": "You can not use this feature because your resolver is set to the string resolver",
  "migrate": "Migrate",
  "reverse_explanation": "The reverse resolution is the procedure to translate an address into a name. It helps to access your address or your domain only knowing one of them after they are associated.",
  "reverse_success": "The reverse resolution has been set.",
  "not_set": "not set",
  "domain": "Domain",
  "register_subdomain": "The subdomain has been created.",
  "update_subdomain": "The subdomain's address has been updated.",
  "remove_subdomain": "The subdomain has been removed.",
  "multi_chain_addresses": "Multi chain addresses",
  "migrate_to_multi_resolver": "The multichain addresses offer domain compatibility with different blockchains. The activation requires a single transaction with your wallet then the multichain addresses will be ready to use.",
  "migration_complete": "Successfully migrated to the multichain resolver.",
  "your_addresses_explanation": "Your domain can be used in different blockchains. Add your addresses for each of the blockchains you want to use.",
  "multichain": "Multichain",
  "chain_address_removed": "Chain address removed",
  "chain_address_updated": "Chain address updated",
  "custom_resolver_message": "This feature will not work because you are using a custom resolver.",
  "activate": "Activate",
  "domain_addresses": "Domain Addresses",
  "public_resolver_explanation": "Your domain is associated to a RSK address by default. You can change it to another RSK address.",
  "change": "Change",
  "services": "Services",
  "whitepaper": "Whitepaper",
  "rif_token": "RIF Token",
  "roadmap": "Roadmap",
  "contact": "Contact",
  "developers": "Developers",
  "privacy": "Privacy",
  "privacy_policy": "Privacy Policy",
  "terms": "Terms & Conditions",
  "get_your_domain": "Get your domain, forget your address",
  "home_explanation": "RIF Name Service provides unique names for your wallet addresses to seamlessly receive crypto assets.",
  "search_placeholder": "Find your domain",
  "ease_of_use": "Ease of use",
  "ease_of_use_explanation": "RIF Name Service (RNS) enables the use of human readable names for blockchain addresses helping users to receive transactions in personalized domains. RNS domains are seamlessly integrated across RIF Services (Payments, Storage, Communications...).",
  "interoperability": "Interoperability",
  "interoperability_explanation": "RNS provides the first multi blockchain naming service available for all major blockchains such as Bitcoin, Ethereum and Litecoin among others. In addition, RNS is built on top of RSK, thus it inherits Bitcoin's decentralized nature and security.",
  "cost_effective": "Cost effective",
  "cose_effective_explanation": "Acquiring a personalized RNS domain for your blockchain address costs less than a dollar. Moreover, the domain owner can create associated subdomains with no additional cost.",
  "are_you_developer": "Are you a developer?",
  "are_you_developer_explanation": "RNS is an open source project which aims to facilitate collaboration and delivers a service with traditional programming language libraries to integrate with different solutions. ",
  "read_documentation": "Read documentation",
  "collaborate_github": "Collaborate on Github",
  "integrate_rns": "Integrate with RNS",
  "ask_question": "Ask us a question",
  "results": "Results",
  "available": "available",
  "search_for_another": "Please search another domain",
  "year": "year",
  "your_domain": "Your domain",
  "enter": "Enter",
  "add_account": "Add account"
}
