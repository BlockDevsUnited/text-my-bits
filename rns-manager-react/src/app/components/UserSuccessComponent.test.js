import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import UserSuccessComponent from './UserSuccessComponent';
import mockStore from '../../../tests/config/mockStore';
import en from '../../languages/en.json';

const store = mockStore({
  close: en.close,
  view_explorer: en.view_explorer,
});

describe('UserSuccessComponent', () => {
  it('renders and matches snapshot', () => {
    const component = mount(<Provider store={store}><UserSuccessComponent /></Provider>);
    expect(component).toMatchSnapshot();
  });

  it('renders correct title and message text', () => {
    const component = mount(
      <Provider store={store}>
        <UserSuccessComponent title="Test Title" message="Test Message!" />
      </Provider>,
    );

    expect(component.find('strong').text()).toEqual('Test Title');
    expect(component.find('p').at(2).text()).toEqual('Test Message!');
  });

  it('renders with explorer address', () => {
    const component = mount(
      <Provider store={store}>
        <UserSuccessComponent title="Test Title" message="Test Message!" address="0x12345" />
      </Provider>,
    );

    expect(component.find('p.explorer').text()).toEqual(en.view_explorer);
  });

  it('returns blank when visible is false', () => {
    const component = mount(
      <Provider store={store}>
        <UserSuccessComponent title="Test Title" message="Test Message!" visible={false} />
      </Provider>,
    );

    expect(component.html()).toEqual('');
  });
});
