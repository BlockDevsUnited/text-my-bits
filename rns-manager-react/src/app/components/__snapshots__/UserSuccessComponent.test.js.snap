// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSuccessComponent renders and matches snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(UserSuccessComponent)>
    <UserSuccessComponent
      address=""
      currentLanguageCode="en"
      dispatch={[Function]}
      handleCloseClick={[Function]}
      message="Success Message"
      strings={
        Object {
          "close": "Close",
          "view_explorer": "View it on the explorer",
        }
      }
      title="Success"
      visible={true}
    >
      <div
        className="success"
      >
        <button
          className="close"
          onClick={[Function]}
          type="button"
        >
          <img
            alt="Close"
            src="mock-image"
          />
        </button>
        <p>
          <img
            alt=""
            src="mock-image"
          />
        </p>
        <p>
          <strong>
            Success
          </strong>
        </p>
        <p>
          Success Message
        </p>
      </div>
    </UserSuccessComponent>
  </Connect(UserSuccessComponent)>
</Provider>
`;
